usr/share/rear/layout/prepare/Linux-s390/400_run_dasd_format_code.sh

```diff
@@ -0,0 +1,185 @@
+                    abort_dasd_format
+                    Error "User did not confirm the recreated DASD format but aborted '$rear_workflow' in ${BASH_SOURCE[0]}"
+                    ;;
+            esac
+        done
+    fi
+    # Run an inner while loop with a user dialog so that the user can fix things
+    # when DASD_FORMAT_CODE failed.
+    # Such a fix does not necessarily mean the user must change
+    # the dasdformat.sh script when DASD_FORMAT_CODE failed.
+    # The user might also fix things by only using the Relax-and-Recover shell and
+    # then confirm what is on the disks and continue with restoring the backup
+    # or abort this "rear recover" run to re-try from scratch.
+    while true ; do
+        choice="$( UserInput -I DASD_FORMAT_CODE_RUN -p "$prompt" -D "${choices[0]}" "${choices[@]}" )" && wilful_input="yes" || wilful_input="no"
+        case "$choice" in
+            (${choices[0]})
+                # Rerun or run (after switching to recreating with DASD format script) DASD format script:
+                is_true "$wilful_input" && LogPrint "User runs DASD format script" || LogPrint "Running DASD format script by default"
+                # Only break the inner while loop (i.e. the user dialog loop):
+                break
+                ;;
+            (${choices[1]})
+                # Run 'less' with the original STDIN STDOUT and STDERR when 'rear' was launched by the user:
+                less $RUNTIME_LOGFILE 0<&6 1>&7 2>&8
+                ;;
+            (${choices[2]})
+                # Run 'vi' with the original STDIN STDOUT and STDERR when 'rear' was launched by the user:
+                vi $DASD_FORMAT_CODE 0<&6 1>&7 2>&8
+                ;;
+            (${choices[3]})
+                LogPrint "This is the current list of DASDs:"
+                lsdasd_output
+                ;;
+            (${choices[4]})
+                # Run 'less' with the original STDIN STDOUT and STDERR when 'rear' was launched by the user:
+                less $original_disk_space_usage_file 0<&6 1>&7 2>&8
+                ;;
+            (${choices[5]})
+                # rear_shell runs 'bash' with the original STDIN STDOUT and STDERR when 'rear' was launched by the user:
+                rear_shell "" "$rear_shell_history"
+                ;;
+            (${choices[6]})
+                # Confirm what is on the disks and continue:
+                # Break the outer while loop and continue with restoring the backup:
+                break 2
+                ;;
+            (${choices[7]})
+                abort_dasd_format
+                Error "User chose to abort '$rear_workflow' in ${BASH_SOURCE[0]}"
+                ;;
+        esac
+    done
+# End of the outer while loop:
+done
+
+# Local functions must be 'unset' because bash does not support 'local function ...'
+# cf. https://unix.stackexchange.com/questions/104755/how-can-i-create-a-local-function-in-my-bashrc
+unset -f lsdasd_output
```

Comment:
not required if you use `local` due to the function scope created by `Source` function

Reply:
required because of what the comment at this code place tells